components/Form/FormURL.tsx

```diff
@@ -28,30 +28,32 @@ interface FormURLProps {
     setFormData(data: Record<string, any>): any;
 }
 const FormURL = ({ id, title, formData, setFormData }: FormURLProps) => {
-    const onSpacePressed = (e: KeyboardEvent<HTMLInputElement>) => {
+    const [error, setError] = useState(""); // Add this line to initialize the error state
+
+    const onSpacePressed = (e: React.KeyboardEvent<HTMLInputElement>) => {
         if (e.key === " ") {
             e.preventDefault();
-            setFormData((data: Record<string, any>) => ({
-                ...data,
-                urls: [...data.urls, (e.target as HTMLInputElement).value.trim()].filter((x) => x.length > 0),
-            }));
-            (e.target as HTMLInputElement).value = "";
-        }
-        if (
-            e.key === "Backspace" &&
-            (e.target as HTMLInputElement).value.length === 0
-        ) {
-            setFormData((data: Record<string, any>) => ({
-                ...data,
-                urls: data.urls.slice(0, -1),
-            }));
+            const inputElement = e.target as HTMLInputElement;
+            const urlToAdd = inputElement.value.trim();
+
+            if (urlToAdd.length > 0 && !formData.urls.includes(urlToAdd)) {
+                setFormData((data: Record<string, any>) => ({
+                    ...data,
+                    urls: [...data.urls, urlToAdd],
+                }));
+                setError(""); // Clear any previous error message
+            } else {
+                setError("Duplicate URL. Please enter a different URL.");
```

Comment:
Pressing space twice (generating a URL of length 0) will bring up the same error message. I think we should either not show an error message, or show a different one